### ISSUE TITLE
Liquibase website is unavailable

### ISSUE THREAD

**Issue by anirudhvsp** [liquibase/liquibase#6364]
### Search first

- [X] I searched and no similar issues were found

### Description

Hi ,
Since today morning IST, [liquibase.org](https://liquibase.org/) is unavailable.
Found similar issue which is an year old (https://github.com/liquibase/liquibase/issues/4355).
This is causing changelog parsing to fail due to a xsd refence in the changelog xml file 
xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd"

### Steps To Reproduce

Visit [liquibase.org](https://liquibase.org/) in a web browser.

### Expected/Desired Behavior

The [liquibase.org](https://liquibase.org/) website loads successfully, specifically http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd 

Workaround (https://github.com/liquibase/liquibase/issues/4355#issuecomment-1578703959)

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR (Thank you!)

**Comment by Stintrist** [comment]
Same is happening here

**Comment by sheik61** [comment]
Please close this issue. 

The old one has been reopened (https://github.com/liquibase/liquibase/issues/4355).

**Comment by locious7** [comment]
Experiencing the same thing.

**Comment by aarif-lumi** [comment]
Same is happening for me.

**Comment by ptpu** [comment]
Same issue as in https://github.com/liquibase/liquibase/issues/4355

**Comment by jeserkin** [comment]
@kevin-atx pinging you, since last time you notified about it being back online.

**Comment by canyuva** [comment]
Is there any solution about reference the xsd file locally instead of remote url ?

`<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
                   https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">`

**Comment by canyuva** [comment]
> > Is there any solution about reference the xsd file locally instead of remote url ?
> > `<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">`
> 
> ([#4355 (comment)](https://github.com/liquibase/liquibase/issues/4355#issuecomment-1578703959))

Sorry but it won't work. Still trying to access to remote url. 

`<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
                   ../dbchangelog-latest.xsd">`
                   
`Caused by: java.net.UnknownHostException: www.liquibase.org
	at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184)
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172)
	at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
	at java.net.Socket.connect(Socket.java:589)
	at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:666)
	at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173)
	at sun.net.NetworkClient.doConnect(NetworkClient.java:180)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:463)
	at sun.net.www.http.HttpClient.openServer(HttpClient.java:558)
	at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264)
	at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1156)
	at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050)
	at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1564)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263)
	at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:647)
	at com.sun.org.apache.xerces.internal.impl.XMLVersionDetector.determineDocVersion(XMLVersionDetector.java:148)
	at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaParsingConfig.parse(SchemaParsingConfig.java:583)
	at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaParsingConfig.parse(SchemaParsingConfig.java:686)
	at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaDOMParser.parse(SchemaDOMParser.java:530)
	at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.getSchemaDocument(XSDHandler.java:2181)`

**Comment by anirudhvsp** [comment]
> > > Is there any solution about reference the xsd file locally instead of remote url ?
> > > `<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">`
> > 
> > 
> > ([#4355 (comment)](https://github.com/liquibase/liquibase/issues/4355#issuecomment-1578703959))
> 
> Sorry but it won't work. Still trying to access to remote url.
> 
> `<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog ../dbchangelog-latest.xsd">`
> 
> It gives this error message :
> 
> `Caused by: java.net.UnknownHostException: www.liquibase.org at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184) at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172) at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) at java.net.Socket.connect(Socket.java:589) at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:666) at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173) at sun.net.NetworkClient.doConnect(NetworkClient.java:180) at sun.net.www.http.HttpClient.openServer(HttpClient.java:463) at sun.net.www.http.HttpClient.openServer(HttpClient.java:558) at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264) at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367) at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191) at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1156) at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050) at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177) at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1564) at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492) at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263) at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:647) at com.sun.org.apache.xerces.internal.impl.XMLVersionDetector.determineDocVersion(XMLVersionDetector.java:148) at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaParsingConfig.parse(SchemaParsingConfig.java:583) at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaParsingConfig.parse(SchemaParsingConfig.java:686) at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaDOMParser.parse(SchemaDOMParser.java:530) at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.getSchemaDocument(XSDHandler.java:2181)`

strange, works for me, however I have used full path of the xsd in the xml,
alternatively, you can also use this url instead of specifying the local directory
(https://raw.githubusercontent.com/liquibase/liquibase/refs/heads/master/liquibase-standard/src/main/resources/www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd)
This also worked for me

**Comment by ijrodrigues** [comment]
This really caught me off guard, the solution that worked here was to upgrade liquibase version from 4.8.0 to 4.29.0

**Comment by anirudhvsp** [comment]
> > > > > Is there any solution about reference the xsd file locally instead of remote url ?
> > > > > `<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog https://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">`
> > > > 
> > > > 
> > > > ([#4355 (comment)](https://github.com/liquibase/liquibase/issues/4355#issuecomment-1578703959))
> > > 
> > > 
> > > Sorry but it won't work. Still trying to access to remote url.
> > > `<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog ../dbchangelog-latest.xsd">`
> > > It gives this error message :
> > > `Caused by: java.net.UnknownHostException: www.liquibase.org at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:184) at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:172) at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392) at java.net.Socket.connect(Socket.java:589) at sun.security.ssl.SSLSocketImpl.connect(SSLSocketImpl.java:666) at sun.security.ssl.BaseSSLSocketImpl.connect(BaseSSLSocketImpl.java:173) at sun.net.NetworkClient.doConnect(NetworkClient.java:180) at sun.net.www.http.HttpClient.openServer(HttpClient.java:463) at sun.net.www.http.HttpClient.openServer(HttpClient.java:558) at sun.net.www.protocol.https.HttpsClient.<init>(HttpsClient.java:264) at sun.net.www.protocol.https.HttpsClient.New(HttpsClient.java:367) at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.getNewHttpClient(AbstractDelegateHttpsURLConnection.java:191) at sun.net.www.protocol.http.HttpURLConnection.plainConnect0(HttpURLConnection.java:1156) at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:1050) at sun.net.www.protocol.https.AbstractDelegateHttpsURLConnection.connect(AbstractDelegateHttpsURLConnection.java:177) at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1564) at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1492) at sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:263) at com.sun.org.apache.xerces.internal.impl.XMLEntityManager.setupCurrentEntity(XMLEntityManager.java:647) at com.sun.org.apache.xerces.internal.impl.XMLVersionDetector.determineDocVersion(XMLVersionDetector.java:148) at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaParsingConfig.parse(SchemaParsingConfig.java:583) at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaParsingConfig.parse(SchemaParsingConfig.java:686) at com.sun.org.apache.xerces.internal.impl.xs.opti.SchemaDOMParser.parse(SchemaDOMParser.java:530) at com.sun.org.apache.xerces.internal.impl.xs.traversers.XSDHandler.getSchemaDocument(XSDHandler.java:2181)`
> > 
> > 
> > strange, works for me, however I have used full path of the xsd in the xml, alternatively, you can also use this url instead of specifying the local directory (https://raw.githubusercontent.com/liquibase/liquibase/refs/heads/master/liquibase-standard/src/main/resources/www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd) This also worked for me. PS: Also make sure the URL is replaced in any other xml files included with `<include>`
> 
> But aren't you this way just trading liquibase.com 3rd party website for https://raw.githubusercontent.com ? If it will be down, then it will also be out of your control.

true, I was just testing performance improvement in changelog parsing when using different origins for this xsd.
turns out the GitHub one is also way faster than the liquibase website. would also recommend self hosting xsd.

**Comment by thangnq1001** [comment]
Below syntax worked for me thanks to @anirudhvsp 's link:
```
<databaseChangeLog
        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog https://raw.githubusercontent.com/liquibase/liquibase/refs/heads/master/liquibase-standard/src/main/resources/www.liquibase.org/xml/ns/dbchangelog/dbchangelog-latest.xsd">
```

**Comment by jeserkin** [comment]
Seems it is back and running.

**Comment by jeserkin** [comment]
Well it is not really closed as if tomorrow it happens again it will be same case all over again. Would be nice to see a permanent solution for this. Otherwise lost time was in vain.

**Comment by anirudhvsp** [comment]
> Well it is not really closed as if tomorrow it happens again it will be same case all over again. Would be nice to see a permanent solution for this. Otherwise lost time was in vain.

As mentioned by @ijrodrigues 

> This really caught me off guard, broke my prod application, the solution that worked here was to upgrade liquibase version from 4.8.0 to 4.29.0, In this version, there are schemas in artifact.

The latest version of liquibase has fallback for local xsd lookup in case of resource access failure.
I don't think it's worth reopening this issue as this was a temporary problem and self hosting xsd seems to be better option anyways.

**Comment by lbadam** [comment]
👋  I'm a PM on the Liquibase team, and I wanted to give a quick update on this issue. This morning we experienced an unforeseen issue with our infrastructure that caused `*.liquibase.com` to be unavailable. The issue has been resolved, and the site is back up. This is not an ongoing issue. 

As stated earlier, newer versions of Liquibase have a fallback to local XSDs when remote options are unavailable so we would encourage you to update Liquibase (latest is 4.29.2).
